tests/integration/test_api/test_config/test_upload_configuration/test_upload_configuration.py

```diff
@@ -0,0 +1,141 @@
+"""
+copyright: Copyright (C) 2015, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: This tests checks that the configuration is set correctly in the manager via the API.
+
+components:
+    - api
+
+suite: config
+
+targets:
+    - manager
+
+daemons:
+    - wazuh-apid
+    - wazuh-modulesd
+    - wazuh-analysisd
+    - wazuh-execd
+    - wazuh-db
+    - wazuh-remoted
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/api/getting-started.html
+    - https://documentation.wazuh.com/current/user-manual/api/configuration.html
+
+tags:
+    - api
+"""
+import pytest
+import requests
+from pathlib import Path
+
+from . import CONFIGURATIONS_FOLDER_PATH, TEST_CASES_FOLDER_PATH
+from wazuh_testing.constants.api import CONFIGURATION_TYPES, MANAGER_CONFIGURATION_ROUTE
+from wazuh_testing.constants.daemons import API_DAEMONS_REQUIREMENTS
+from wazuh_testing.modules.api.utils import login, get_base_url
+from wazuh_testing.utils.configuration import get_test_cases_data, load_configuration_template
+
+
+# Marks
+pytestmark = pytest.mark.server
+
+# Variables
+# Used by add_configuration to select the target configuration file
+configuration_type = CONFIGURATION_TYPES[0]
+
+# Paths
+test_configuration_path = Path(CONFIGURATIONS_FOLDER_PATH, 'configuration_upload_configuration.yaml')
+test_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_upload_configuration.yaml')
+
+# Configurations
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+daemons_handler_configuration = {'daemons': API_DAEMONS_REQUIREMENTS}
+
+# Tests
+@pytest.mark.tier(level=0)
+@pytest.mark.parametrize('test_configuration,test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_upload_configuration(test_configuration, test_metadata, add_configuration, truncate_monitored_files,
+                              daemons_handler, wait_for_api_start):
+    """
+    description: Check if the API works when uploading configurations.
+
+    wazuh_min_version: 4.2.0
```

Comment:
The limit configuration related to the EPS was added in #13608. The Wazuh minimum version should be changed since the base conf includes it.